libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -823,6 +836,32 @@ void CompilerUtils::cleanHigherOrderBits(IntegerType const& _typeOnStack)
 		m_context << ((u256(1) << _typeOnStack.numBits()) - 1) << Instruction::AND;
 }
 
+void CompilerUtils::cleanHigherOrderBits(FixedPointType const& _typeOnStack)
+{
+	if (_typeOnStack.numBits() == 256)
+		return;
+	else if (_typeOnStack.isSigned())
```

Comment:
Just don't duplicate code - call the integer function here with an integer type that has the same cleaning behaviour.